packages/cubejs-cockroach-driver/src/CockroachDriver.ts

```diff
@@ -0,0 +1,275 @@
+import { types, Pool, PoolConfig, PoolClient, FieldDef } from 'pg';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { TypeId, TypeFormat } from 'pg-types';
+import { getEnv } from '@cubejs-backend/shared';
+import * as moment from 'moment';
+import {
+  BaseDriver,
+  DownloadTableMemoryData, DriverInterface,
+  GenericDataBaseType, IndexesSQL, TableStructure, QueryOptions,
+} from '@cubejs-backend/query-orchestrator';
+
+const GenericTypeToCockroach: Record<GenericDataBaseType, string> = {
+  double: 'decimal',
+  // Revert mapping for internal pre-aggregations
+  HLL_POSTGRES: 'hll',
+};
+
+const NativeTypeToPostgresType: Record<string, string> = {};
+
+Object.entries(types.builtins).forEach(([key, value]) => {
+  NativeTypeToPostgresType[value] = key;
+});
+
+const PostgresToGenericType: Record<string, GenericDataBaseType> = {
+  // bpchar (“blank-padded char”, the internal name of the character data type)
+  bpchar: 'varchar',
+  // Numeric is an alias
+  numeric: 'decimal',
+  // External mapping
+  hll: 'HLL_POSTGRES',
+};
+
+const timestampDataTypes = [
+  // @link TypeId.DATE
+  1082,
+  // @link TypeId.TIMESTAMP
+  1114,
+  // @link TypeId.TIMESTAMPTZ
+  1184
+];
+const timestampTypeParser = (val: string) => moment.utc(val).format(moment.HTML5_FMT.DATETIME_LOCAL_MS);
+const hllTypeParser = (val: string) => Buffer.from(
+  // Postgres uses prefix as \x for encoding
+  val.slice(2),
+  'hex'
+).toString('base64');
+
+export type PostgresDriverConfiguration = Partial<PoolConfig> & {
+  storeTimezone?: string,
+  executionTimeout?: number,
+  readOnly?: boolean,
+};
+
+export class CockroachDriver<Config extends PostgresDriverConfiguration = PostgresDriverConfiguration>
```

Comment:
According to the latest changes, you need to specify the static `getDefaultConcurrency` method for the driver. You can check it with any other driver. But in general it should looks like this:

```
  /**
   * Returns default concurrency value.
   */
  public static getDefaultConcurrency(): number {
    return 2;
  }
```
Returned value could be `2` for now.